x-pack/plugins/cases/public/components/edit_connector/index.test.tsx

```diff
@@ -361,4 +361,18 @@ describe('EditConnector ', () => {
       expect(result.queryByTestId('has-data-to-push-button')).toBe(null);
     });
   });
+
+  it('does not show the edit connectors pencil if the user does not have read access to actions', async () => {
+    const defaultProps = getDefaultProps();
+    const props = { ...defaultProps, connectors: [] };
+    appMockRender.coreStart.application.capabilities = {
+      ...appMockRender.coreStart.application.capabilities,
+      actions: { save: false, show: false },
+    };
+
+    const result = appMockRender.render(<EditConnector {...props} />);
+    await waitFor(() => {
+      expect(result.queryByTestId('connector-edit')).toBe(null);
```

Comment:
This test might not be testing anything.

if this function runs before the component gets the chance to render, this will immediately pass and if in the future something changes and this element is back to render, this test won't catch it.

I advice you check here also for an element that you now must be rendered and then you check for the one that should not render. That way you are sure the null you get is because the element is not rendered and not because the component did not render.

Reply:
Thank you @academo! Very good observation. I will change it accordingly.